docs: added schematic test info to tests docs (#28)
terraform-ibm-modules/documentation: docs/tests.md

Tests are written in the Go programming language to take advantage of an open source Go library called [Terratest](https://github.com/gruntwork-io/terratest).

> The tests also use test helper functions from [ibmcloud-terratest-wrapper](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper) ([source](https://github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper)).

Reviewer: ```suggestion
The tests also use test helper functions from the [ibmcloud-terratest-wrapper](https://github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper) module. For more information about the functions, see the Go package at [pkg.go.dev](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper)
```

how about this? trying to keep the links and link text consistent with the rest of the docs.

  ?> **Tip**: The upgrade test is disabled by default in `pr_test.go` in the module template because you can't run an upgrade test until the initial module code is merged to the main branch. After the initial merge, create a pull request to enable the upgrade test by commenting out the line that starts with `t.Skip`.

- **IBM Cloud Schematics test**

>     Use the `testschematic/TestSchematicOptions.RunSchematicTest()` function to execute a test example from within an IBM Cloud Schematics Workspace.

Reviewer: ```suggestion
    Use the `testschematic/TestSchematicOptions.RunSchematicTest()` function to execute a test example from within an IBM Cloud Schematics Workspace. The function includes the following things:
```

`things` is not great. got another words the categorizes the list?

>     - Sets up a new transient workspace in your account to execute tests
>     - Configurable tar file is built that is uploaded to workspace
>     - `init`, `apply`, and `destroy` actions for workspace are executed
>     - Workspace is deleted on successful test
>     - Supports setting of Workspace environment variables
>     - Supports Schematic Variablestore settings for test input variables
>     - netrc settings can be configured to support terraform modules in private github source locations

Reviewer: ```suggestion
    - Sets up a new transient workspace in your account to execute tests
    - Builds a configurable .tar file that is uploaded to the workspace
    - Executes `init`, `apply`, and `destroy` actions for the workspace
    - Deletes the workspace after a successful test
    - Supports setting workspace environment variables
    - Supports Schematics `variablestore` settings for test input variables
    - Allows `.netrc` settings to support Terraform modules in private GitHub source locations
```

Making the list items parallel by starting with a verb.

> [Examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#pkg-overview) can be found at offical repository.

Reviewer: ```suggestion
For more information, see the [examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#pkg-overview) at pkg.go.dev.
```

## Other test customizations

> ### Review all test options available
> 
> The test option objects `TestOptions` and `TestSchematicOptions` have many fields that can be used to further configure your test runs. You can review all of the current options available at the official Go package repository for `ibmcloud-terratest-wrapper`.
> - [TestOptions](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper#TestOptions)
> - [TestSchematicOptions](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#TestSchematicOptions)
> 

Reviewer: ```suggestion
### TestOptions and TestSchematicOptions

You can configure you tests by using the fields in the `TestOptions` and `TestSchematicOptions` objects:

- [TestOptions](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper#TestOptions)
- [TestSchematicOptions](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#TestSchematicOptions)

```

For example, if a test spins up a Kubernetes cluster and then deploys things to it, you can use `ImplicitDestroy` to remove all the resources that are deployed. The resources are destroyed along with the cluster deletion.

!> **Important** Never pass any resources to `ImplicitDestroy` that are created as part of the module that you are testing. The tests are designed to follow the same flow as a user who runs `terraform destroy`. Use this feature only on supporting resources of the module that are used in the end-to-end test.

> ### More Examples
> 
> For more examples of various test customization, you can consult the official Go package repository for `ibmcloud-terratest-wrapper`.
> - [Terratest Examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper#pkg-overview)
> - [Schematics Workspace Examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#pkg-overview)
> - [README Examples](https://github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper#readme)

Reviewer: ```suggestion
### More examples

For more examples of test customization, see the pkg.go.dev repository and the GitHub repo for `ibmcloud-terratest-wrapper`:

- [Terratest examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper#pkg-overview)
- [Schematics Workspace examples](https://pkg.go.dev/github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testschematic#pkg-overview)
- [Usage examples](https://github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper) in the GitHub repo.
```
